internal/action/command.go

```diff
@@ -854,7 +854,7 @@ func (h *BufPane) ReplaceCmd(args []string) {
 					nreplaced++
 				} else if !canceled && !yes {
 					searchLoc = locs[0]
-					searchLoc.X += util.CharacterCount(replace)
+					searchLoc.X = locs[1].X
```

Comment:
`X` now comes from `locs[1]`, but `Y` is taken from `locs[0]`. I'm not sure if this is okay.

Reply:
I think it will work fine, since `locs[0].Y` is always equal to `locs[1].Y` in this case (since micro doesn't currently support multi-line search matches). But for code clarity it's better to replace these 2 lines with `searchLoc = locs[1]`.

Overall, this is a good catch. Using `util.CharacterCount(replace)` here is wrong not only in the case of empty replace. For example, with `replace [a-z][a-z] xxxxxxxx` saying "no" will not move to the next closest match but will "jump over" a couple of next matches (it can be easily seen if `hlsearch` option is on), since the length of the replacement string has nothing to do with the length of the matching string.